app/src/main/java/com/hamdan/forzenbook/viewmodels/SearchViewModel.kt

```diff
@@ -0,0 +1,29 @@
+package com.hamdan.forzenbook.viewmodels
+
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import com.hamdan.forzenbook.search.core.domain.GetPostByStringUseCase
+import com.hamdan.forzenbook.search.core.domain.GetPostByUserIdUseCase
+import com.hamdan.forzenbook.search.core.viewmodel.BaseSearchViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class SearchViewModel @Inject constructor(
+    private val postById: GetPostByUserIdUseCase,
+    private val postByQuery: GetPostByStringUseCase,
+) : BaseSearchViewModel(
+    postByQuery, postById
+) {
+
+    private val _state: MutableState<SearchState> =
+        mutableStateOf(SearchState.Content(listOf(), ""))
```

Comment:
Make default arguments for Content so you don't need these here

Reply:
Big true, biiiiiig true